eslint/babel-eslint-parser/src/convert/convertAST.js

```diff
@@ -81,30 +81,30 @@ function convertNodes(ast, code) {
     },
   };
   const state = { source: code };
-
-  const oldExportAllDeclarationKeys = t.VISITOR_KEYS.ExportAllDeclaration;
+  const oldVisitorKeys = new Map();
 
   try {
-    // Monkey patch visitor keys in order to be able to traverse the estree nodes
-    t.VISITOR_KEYS.ChainExpression = VISITOR_KEYS.ChainExpression;
-    t.VISITOR_KEYS.ImportExpression = VISITOR_KEYS.ImportExpression;
-    t.VISITOR_KEYS.Property = VISITOR_KEYS.Property;
-    t.VISITOR_KEYS.MethodDefinition = VISITOR_KEYS.MethodDefinition;
+    for (const [type, visitorKey] of Object.entries(conflictTypes)) {
```

Comment:
Also refactored a bit to remove boilerplates.

Reply:
Nice refactor!